src/sinks/s3_common/partitioner.rs

```diff
@@ -0,0 +1,60 @@
+use vector_core::{event::Event, partition::Partitioner};
+
+use crate::{internal_events::TemplateRenderingError, template::Template};
+
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub struct S3PartitionKey {
+    pub key_prefix: String,
+    pub ssekms_key_id: String,
```

Comment:
I think it's best to keep this as an `Option` rather than represent an `None` template as `""`.

```suggestion
    pub ssekms_key_id: Option<String>,
```

Reply:
Agreed...that looks less awkward 👍